lib/http.php

```diff
@@ -77,24 +79,40 @@ public function request(string $url, array $config = []): Response
             'max_filesize' => null,
             'max_redirections' => 5,
         ];
+
+        // if curl-impersonate is not detected, use some basic defaults
+        if (curl_version()['ssl_version'] != 'BoringSSL') {
+            // Snagged from https://github.com/lwthiker/curl-impersonate/blob/main/firefox/curl_ff102
+            $defaults['headers'] = [
+                'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
+                'Accept-Language' => 'en-US,en;q=0.5',
+                'Upgrade-Insecure-Requests' => '1',
+                'Sec-Fetch-Dest' => 'document',
+                'Sec-Fetch-Mode' => 'navigate',
+                'Sec-Fetch-Site' => 'none',
+                'Sec-Fetch-User' => '?1',
+                'TE' => 'trailers',
+            ];
+            curl_setopt($ch, CURLOPT_HEADER, false);
+            curl_setopt($ch, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
```

Comment:
Do these two options strictly need to be not set for curl-impersonate usage?

Beware that `CURLOPT_PROTOCOLS` is very important for security.

Reply:
If you set CURLOP_HEADER to false, it will negate all that curl-impersonate does internally. However, I think you're right on `CURLOPT_PROTOCOLS`, I assumed this could have affected the usage of HTTP2 somehow, as the latter is something that is used to fingerprint webbrowsers and `libcurl-impersonate` explicitly toggles on/off as needed, but it is not what `CURLOPT_PROTOCOLS` does.